File: ResourceManager/ViewModels/Articles/ArticlePageViewModel.vb
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Security
Imports ResourceManager.Configuration
Imports ResourceManager.Helpers
Imports System.Collections.ObjectModel
Imports System.Linq
Imports Microsoft.Extensions.DependencyInjection

Namespace ResourceManager.ViewModels
    Public Class ArticlePageViewModel
        Inherits PageViewModel
        Public Sub New()

            PageTitle = "Article"
            PageIcon = AssetManager.Instance.GetImage("Articles.png")
            PageColor = ArticlePageColor
            MyBase.HasPermission = AuthManager.User.IsInPermission(VIEW_ARTICLE_PERMISSION)

            Dim tagsContent = Application.ServiceProvider.GetRequiredService(Of TagsContentViewModel)()
            tagsContent.PageColor = PageColor
            tagsContent.ArticleColumnWidth = 100

            MenuItems = New ObservableCollection(Of PageContentViewModel) From {
    Application.ServiceProvider.GetRequiredService(Of ArticleContentViewModel)(),
    tagsContent,
    Application.ServiceProvider.GetRequiredService(Of ArticleOptionsContentViewModel)()
}

            SelectedItem = MenuItems.FirstOrDefault()
        End Sub
    End Class
End Namespace

File: ResourceManager/ViewModels/Articles/ArticlePageViewModel.cs
using Microsoft.Extensions.DependencyInjection;
using NetCore.Mvvm.Abstractions;
using NetCore.Security;
using ResourceManager.Configuration;
using ResourceManager.Helpers;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace ResourceManager.ViewModels
{
    public class ArticlePageViewModel : PageViewModel
    {
        public ArticlePageViewModel()
        {

            PageTitle = "Article";
            PageIcon = AssetManager.Instance.GetImage("Articles.png");
            PageColor = Constants.ArticlePageColor;
            HasPermission = AuthManager.User.IsInPermission(Constants.VIEW_ARTICLE_PERMISSION);

            var tagsContent = App.ServiceProvider.GetRequiredService<TagsContentViewModel>();
            tagsContent.PageColor = PageColor;
            tagsContent.ArticleColumnWidth = 100;

            MenuItems = new ObservableCollection<PageContentViewModel>
            {
                App.ServiceProvider.GetRequiredService<ArticleContentViewModel>(),
                tagsContent,
                App.ServiceProvider.GetRequiredService<ArticleOptionsContentViewModel>()
            };
            
            SelectedItem = MenuItems.FirstOrDefault();
        }
    }
}
